calour/dsfdr.py

```diff
@@ -114,106 +123,115 @@ def dsfdr(data, labels, method='meandiff', transform='rankdata', alpha=0.1, nump
         'pearson' : pearson correlation (numeric)
         'nonzerospearman' : spearman correlation only non-zero entries (numeric)
         'nonzeropearson' : pearson correlation only non-zero entries (numeric)
-        function : use this function to calculate the t-statistic (input is data,labels, output is array of float)
-
-    transform : str or ''
-        transformation to apply to the data before caluculating the statistic
-        'rankdata' : rank transfrom each OTU reads
-        'log2data' : calculate log2 for each OTU using minimal cutoff of 2
-        'normdata' : normalize the data to constant sum per samples
-        'binarydata' : convert to binary absence/presence
+        function : use this function to calculate the test statistic (input is data,labels, output is array of float)   
 
     alpha : float
         the desired FDR control level
     numperm : int
         number of permutations to perform
 
+    fdrmethod : str
+        the FDR procedure to determine significant bacteria
+        'dsfdr' : discrete FDR method
+        'bhfdr' : Benjamini-Hochberg FDR method
+        'byfdr' : Benjamini-Yekutielli FDR method 
+        'filterBH' : Benjamini-Hochberg FDR method with filtering
+
     output:
     reject : np array of bool (length N)
         True for OTUs where the null hypothesis is rejected
     tstat : np array of float (length N)
-        the t-statistic value for each OTU (for effect size)
+        the test statistic value for each OTU (for effect size)
     pvals : np array of float (length N)
-        the p-value for each OTU
+        the p-value for each OTU    
     '''
+    
     logger.debug('dsfdr using fdr method: %s' % fdrmethod)
-    data = data.copy()
+
+    data=data.copy()
+
+    if fdrmethod == "filterBH":
+        index=[]
+        n0 = np.sum(labels==0)
+        n1 = np.sum(labels==1)
+
+        for i in range(np.shape(data)[0]):
+            nonzeros = np.count_nonzero(data[i,:])
+            if nonzeros < min(n0,n1):
+                pval_min = (comb(n0, nonzeros, exact=True) + comb(n1, nonzeros, exact=True))/comb(n0+n1, nonzeros)
+                if pval_min <= alpha:
+                    index.append(i)  
+            else:
+                index.append(i)
+        data = data[index,:]    
+
+
     # transform the data
     if transform == 'rankdata':
```

Comment:
here it is 'transform' instead of 'transform_type'? does it pass unit testing (i.e. running nosetests)?